packages/cli/src/credentials/credentials.service.ts

```diff
@@ -36,4 +96,162 @@ export class CredentialsService {
 		}
 		return new Credentials({ id: id.toString(), name }, type, nodesAccess, data);
 	}
+
+	static async prepareCreateData(
+		data: CredentialRequest.CredentialProperties,
+	): Promise<CredentialsEntity> {
+		// Make a copy so we can delete the provided ID and
+		// so we can modify nodesAccess
+		const dataCopy = clone(data);
+		delete dataCopy.id;
+
+		const newCredentials = new CredentialsEntity();
+		Object.assign(newCredentials, dataCopy);
+
+		await validateEntity(newCredentials);
+
+		// Add the date for newly added node access permissions
+		for (const nodeAccess of newCredentials.nodesAccess) {
+			nodeAccess.date = new Date();
+		}
+
+		return newCredentials;
+	}
+
+	static async prepareUpdateData(
+		data: CredentialRequest.CredentialProperties,
+		decryptedData: ICredentialDataDecryptedObject,
+	): Promise<CredentialsEntity> {
+		const updateData = new CredentialsEntity();
+		Object.assign(updateData, data);
```

Comment:
same as previous comment
```suggestion
		const updateData = Db.collections.Credentials.create(data);
```